apps/experiments/views/experiment.py

```diff
@@ -1271,6 +1271,50 @@ def _experiment_chat_ui(request, embedded=False):
     )
 
 
+@experiment_session_view()
+@verify_session_access_cookie
+def experiment_session_messages_view(request, team_slug: str, experiment_id: uuid.UUID, session_id: str):
+    """View for loading paginated messages with HTMX"""
+    session = request.experiment_session
+    experiment = request.experiment
+    page = int(request.GET.get("page", 1))
+    search = request.GET.get("search", "")
+    page_size = 100
+    all_messages = session.get_messages_for_display()
+
+    if search:
+        filtered_messages = [
+            msg for msg in all_messages if any(search.lower() in tag.lower() for tag in msg.all_tag_names())
```

Comment:
Feels like this can be done with a query?

Reply:
+1